src/sdl/sdlimageloader.cpp

```diff
@@ -87,14 +87,23 @@ namespace gcn
                     std::string("Not enough memory to load: ") + filename);
         }
 
-        Image *image = new SDLImage(surface, true, mRenderer);
+        mImage = new SDLImage(surface, true, mRenderer);
 
         if (convertToDisplayFormat)
         {
-            image->convertToDisplayFormat();
+            mImage->convertToDisplayFormat();
         }
 
-        return image;
+        return mImage;
```

Comment:
Purpose of `ImageLoader` is to load several images, Here you need one imageLoader by image...
For RAII, it should return a smart pointer instead...
Unsure about minimal standard wanted for guisan though...

Reply:
I made these changes to `ImageLoader` to fix a memory leak problem I noticed (with Valgrind). The images allocated were never freed again, and if you loaded several in your application, the lost memory increased. Perhaps there's a better way of doing it, but this worked for me at least :) Open to suggestions.

Reply:
As I said, the proper fix would be to return `std::unique_ptr<gcn::Image>`, but it requires C++11 (and changes API).
Guisan has 2 usages of C++11 `auto`, so minimum standard version might be C++11... (but there is still API changes...)

Internal usage of `Image::load` (`Icon`, `ImageButton`, `ImageFont`) does the deletion.
Examples misuse it though.
Your changes would produce double free (unless you also modify those classes).

Reply:
Sorry for the delay, I'm having some health issues that make it very painful to use a computer the last few weeks.

I've added a `free` function as well, which can be used to delete the image when you're done with it. I call that from my code, when cleaning up. The examples here haven't been updated, as that was not in my scope during this PR (though I could modify them accordingly as well).

I don't see how this approach would lead to double free errors - indeed, I guess I would have run into them in my app, if that was the case. Perhaps I'm missing something you're seeing, but this approach seems to work fine for my needs. Could you elaborate more, please?

Reply:
>  as that was not in my scope during this PR (though I could modify them accordingly as well).

As I said, split that PR to be manageable, instead of continuing to make the PR huge. ;-)
Note that I am not a maintainer, but a contributor like you :).

> I don't see how this approach would lead to double free errors 

If you look at `Icon`:

```
    Icon::Icon(const std::string& filename)
    {
        mImage = Image::load(filename);
        mInternalImage = true;
        setHeight(mImage->getHeight());
        setWidth(mImage->getWidth());
    }

    Icon::~Icon()
    {
        if (mInternalImage)
        {
            delete mImage;
        }
    }
```
It **does** the deletion.

`ImageLoader` is a factory, so, if you want auto deletion, you would need a vector of all images created (or returning a smart pointer) and so fix usages which does the deletion.